packages/library-base/src/commonMain/kotlin/io/realm/kotlin/Realm.kt

```diff
@@ -118,9 +118,11 @@ public interface Realm : TypedRealm {
             }
             if (!fileExists(configuration.path)) return false
             val config = (configuration as InternalConfiguration)
-            val (dbPointer, _) = RealmInterop.realm_open(config.createNativeConfiguration())
+            val scheduler = RealmInterop.realm_create_scheduler()
+            val (dbPointer, _) = RealmInterop.realm_open(config.createNativeConfiguration(), scheduler)
             return RealmInterop.realm_compact(dbPointer).also {
                 RealmInterop.realm_close(dbPointer)
+                scheduler.release()
```

Comment:
We will not correctly release the scheduler if `realm_open()` throws an exception (which there is a high chance it will do). There is also a risk if `realm_close()` fails, but I'm not sure that can happen without something really weird being wrong.